src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -244,6 +260,32 @@ internal sealed class CompilerAnalyzerAssemblyResolver(AssemblyLoadContext compi
                 }
             }
         }
+
+        private sealed class DiskResolver : IAnalyzerAssemblyResolver
+        {
+            public static readonly IAnalyzerAssemblyResolver Instance = new DiskResolver();
+            public Assembly? Resolve(AnalyzerAssemblyLoader loader, AssemblyName assemblyName, AssemblyLoadContext directoryContext, string directory)
+            {
+                var assemblyPath = loader.GetAssemblyLoadPath(assemblyName, directory);
+                return assemblyPath is not null ? directoryContext.LoadFromAssemblyPath(assemblyPath) : null;
+            }
+        }
+
+        private sealed class StreamResolver : IAnalyzerAssemblyResolver
```

Comment:
Not sure if you want to have some comments here for the history of why we have these two implementations. I know since I recall your discussions over them, but not sure if they're commented elsewhere.